app/main.py

```diff
@@ -1,25 +1,70 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Visitor, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __set__(self, instance: SlideLimitationValidator, value: int) -> None:
+        setattr(
+            instance,
+            self.protected_name,
+            self.min_amount <= value <= self.max_amount
+        )
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: SlideLimitationValidator
+    ) -> None:
+        return getattr(instance, self.protected_name)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    weight = IntegerRange(20, 50)
+    height = IntegerRange(80, 120)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    weight = IntegerRange(50, 120)
+    height = IntegerRange(120, 220)
 
 
 class Slide:
-    pass
+    def __init__(
+            self,
+            name: str,
+            limitation_class: type[SlideLimitationValidator]
+    ) -> None:
+        self.name = name
+        self.limitation_class = limitation_class
+
+    def can_access(self, visitor: Visitor) -> bool:
+        check = self.limitation_class(
+            age=visitor.age,
+            weight=visitor.weight,
+            height=visitor.height
+        )
+        return all([check.age, check.weight, check.height])
```

Comment:
use `try: .... except`  here

```suggestion
        try:
             self.limitation_class(
                 visitor.age,
                 visitor.height,
                 visitor.weight
             )
         except (ValueError, TypeError):
             return False
         return True
```

Reply:
Okay, will do. However, it was not mentioned in ReedMe that it has to be handled with `try: ... except:`
What are the benefits of using `try: ... except:` here in comparison with original solution?